test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -1,138 +1,177 @@
+
+        davidElliot    = new Person(new Name("David Elliot"),
+                                    new Phone("84512575", false),
+                                    new Email("douglas@nuscomputing.com", false),
+                                    new Address("11 Arts Link", false),
+                                    new HashSet<>(Arrays.asList(tagEconomist, tagPrizeWinner)));
+
+        emptyAddressBook = new AddressBook();
+        defaultAddressBook = new AddressBook(new UniquePersonList(aliceBetsy, bobChaplin));
+    }
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void addPerson_personAlreadyInList_throwsDuplicatePersonException() throws Exception {
+        thrown.expect(DuplicatePersonException.class);
+        defaultAddressBook.addPerson(aliceBetsy);
+    }
+
+    @Test
+    public void containsPerson() throws Exception {
+        UniquePersonList personsWhoShouldBeIn = new UniquePersonList(aliceBetsy, bobChaplin);
+        UniquePersonList personsWhoShouldNotBeIn = new UniquePersonList(charlieDouglas, davidElliot);
+
+        for (Person personWhoShouldBeIn : personsWhoShouldBeIn) {
+            assertTrue(defaultAddressBook.containsPerson(personWhoShouldBeIn));
+        }
+        for (Person personWhoShouldNotBeIn : personsWhoShouldNotBeIn) {
+            assertFalse(defaultAddressBook.containsPerson(personWhoShouldNotBeIn));
+        }
+
+        UniquePersonList allPersons = new UniquePersonList(aliceBetsy, bobChaplin, charlieDouglas, davidElliot);
+
+        for (Person person : allPersons) {
+            assertFalse(emptyAddressBook.containsPerson(person));
+        }
+    }
+
+    @Test
+    public void removePerson_personExists_removesNormally() throws Exception {
+        int numberOfPersonsBeforeRemoval = getSize(defaultAddressBook.getAllPersons());
+        defaultAddressBook.removePerson(aliceBetsy);
+
+        assertFalse(defaultAddressBook.containsPerson(aliceBetsy));
+
+        int numberOfPersonsAfterRemoval = getSize(defaultAddressBook.getAllPersons());
+        assertTrue(numberOfPersonsAfterRemoval == numberOfPersonsBeforeRemoval - 1);
+
+    }
+
+    @Test
+    public void removePerson_personNotExists_throwsPersonNotFoundException() throws Exception {
+        thrown.expect(PersonNotFoundException.class);
+        defaultAddressBook.removePerson(charlieDouglas);
+    }
+
+    @Test
+    public void clear() {
+        defaultAddressBook.clear();
+
+        assertTrue(isEmpty(defaultAddressBook.getAllPersons()));
+    }
+
+    @Test
+    public void getAllPersons() throws Exception {
+        UniquePersonList allPersons = defaultAddressBook.getAllPersons();
+        UniquePersonList personsToCheck = new UniquePersonList(aliceBetsy, bobChaplin);
+
+        assertTrue(isIdentical(allPersons, personsToCheck));
+    }
+
+    @Test
+    public void getAllSortedPersons() throws Exception {
+        AddressBook book = new AddressBook();

```

Comment:
Do you need to create new instances of the address book? What happened in the set up for the test?